src/java/org/apache/cassandra/index/sai/disk/v1/bbtree/BlockBalancedTreeWalker.java

```diff
@@ -170,60 +242,18 @@ final static class TraversalState
         @VisibleForTesting
         int maxLevel;
 
-        TraversalState(DataInput dataInput) throws IOException
+        private TraversalState()
         {
-            maxPointsInLeafNode = dataInput.readVInt();
-            bytesPerValue = dataInput.readVInt();
-
-            // Read index:
-            numLeaves = dataInput.readVInt();
-            assert numLeaves > 0;
-            treeDepth = dataInput.readVInt();
-            minPackedValue = new byte[bytesPerValue];
-            maxPackedValue = new byte[bytesPerValue];
-
-            dataInput.readBytes(minPackedValue, 0, bytesPerValue);
-            dataInput.readBytes(maxPackedValue, 0, bytesPerValue);
-
-            if (ByteArrayUtil.compareUnsigned(minPackedValue, 0, maxPackedValue, 0, bytesPerValue) > 0)
-            {
-                String message = String.format("Min packed value %s is > max packed value %s.",
-                                               new BytesRef(minPackedValue), new BytesRef(maxPackedValue));
-                throw new CorruptIndexException(message, dataInput);
-            }
-
-            valueCount = dataInput.readVLong();
-
-            int numBytes = dataInput.readVInt();
-            byte[] packedIndex = new byte[numBytes];
-            dataInput.readBytes(packedIndex, 0, numBytes);
-
             nodeID = 1;
             level = 0;
             leafBlockFPStack = new long[treeDepth];
             leftNodePositions = new int[treeDepth];
             rightNodePositions = new int[treeDepth];
             splitValuesStack = new byte[treeDepth][];
-
-            memoryUsage = ObjectSizes.sizeOfArray(packedIndex) +
-                          ObjectSizes.sizeOfArray(minPackedValue) +
-                          ObjectSizes.sizeOfArray(maxPackedValue) +
-                          ObjectSizes.sizeOfArray(leafBlockFPStack) +
-                          ObjectSizes.sizeOfArray(leftNodePositions) +
-                          ObjectSizes.sizeOfArray(rightNodePositions) +
-                          ObjectSizes.sizeOfArray(splitValuesStack) * bytesPerValue;
```

Comment:
So we're no longer counting the `TraversalState` items in `memoryUsage` in the walker, but that's okay because the bits in state are transient/query-bound anyway?

Reply:
Exactly that. We are only meant to report the memory used by the long-term objects, which in this case is the `packedIndex`.